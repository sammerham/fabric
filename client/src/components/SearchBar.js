import { React, useState, useContext } from 'react'

import MediaContext from '../mediaContext';
import {
  StyledSearchInput,
  StyledSearchForm,
  StyledSearchButton
} from './StyledComponents'
import MediaApi from '../api/mediaApi';

/*
 * SearchBar
 *
 *  This is the page where you rednder search bar and submit button.
 * State: None

 * props:
      None
  
 * App ---->> GroupButton ---->> SearchBar
 */

function SearchBar() {
  const [formData, setFormData] = useState('');
  
  const { setData } = useContext(MediaContext);
  
  const { getMedia } = MediaApi;
  

  // handleChange with form Data
  const handleChange = e => {
    const { value } = e.target;
    setFormData(data => value)
  };



  //fn fetch new media data 
  async function fetchMatrixData(formData) {
    const MatrixData = await getMedia(formData);
      setData(data => MatrixData.Search);
  };

  
  // handleSubmit when form Submits
  const handleSubmit = e => {
    e.preventDefault();
    if (formData)
      fetchMatrixData(formData)
    setFormData("");
  };


  return (
    <div>
      <StyledSearchForm onSubmit={handleSubmit}>
        <StyledSearchInput
          type="text"
          name="term"
          id="term"
          placeholder="Enter a search Term"
          value={formData}
          onChange={handleChange}
        />
        <StyledSearchButton bg={'#58755b'} onClick={handleSubmit}>Search</StyledSearchButton>
      </StyledSearchForm>
    </div>
  )
}

export default SearchBar
