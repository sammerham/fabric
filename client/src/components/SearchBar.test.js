import React from 'react'
import { render } from '@testing-library/react';
import SearchBar from './SearchBar';
import MediaContext from '../mediaContext';

const MOCK = () => console.log('hello')


// smoke test
it('renders without crashing', () => {
  <SearchBar />
});

// snapShot
it("matches snapshot", function () {
  const { container } = render(
    <MediaContext.Provider value={{ MOCK }}>
      <SearchBar />
    </MediaContext.Provider>
  );
  expect(container).toMatchSnapshot();
});