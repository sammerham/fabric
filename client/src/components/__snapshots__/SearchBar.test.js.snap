// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div>
    <form
      class="sc-kDTinF jSvsQF"
    >
      <input
        class="sc-pVTFL xFLhP"
        id="term"
        name="term"
        placeholder="Enter a search Term"
        type="text"
        value=""
      />
      <button
        class="sc-jrQzAO bGNTCT"
      >
        Search
      </button>
    </form>
  </div>
</div>
`;
